aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -705,6 +705,64 @@ struct ReductionMulOp {
   __forceinline__ scalar_t identity_cpu() const { return 1; }
 };
 
+void _apply_sparse_csr_linear_solve(
+  const Tensor& A,
+  const Tensor& b,
+  const bool left,
+  const Tensor& x) {
+#if defined(USE_ROCM) || !defined(USE_CUDSS)
+  TORCH_CHECK(
+      false,
+      "Calling linear solver with sparse tensors requires compiling ",
+      "PyTorch with CUDA cuDSS and is not supported in ROCm build.");
+#else
+  TORCH_CHECK(A.is_sparse_csr(), "A must be a CSR matrix");
+  TORCH_CHECK(b.dim() == 1, "b must be a 1D tensor");
+  TORCH_CHECK(b.stride(0) == 1, "b must be a column major tensor");
+  TORCH_CHECK(b.size(0) == A.size(-1), "linear system size mismatch.");
+  TORCH_CHECK(A.dtype() == b.dtype(), "A and b must have the same dtype");
+  TORCH_CHECK(left == true, "only left == true is supported by the Sparse CSR backend")
+
+  Tensor crow = A.crow_indices();
+  Tensor col = A.col_indices();
+  if (crow.scalar_type() != ScalarType::Int) {
+    crow = crow.to(crow.options().dtype(ScalarType::Int));
+    col = col.to(col.options().dtype(ScalarType::Int));
+  }
+  int* rowOffsets = crow.data<int>();
+  int* colIndices = col.data<int>();
+  Tensor values = A.values();
+  // cuDSS data structures and handle initialization
+  cudssConfig_t config;
+  cudssMatrix_t b_mt;
+  cudssMatrix_t A_mt;
+  cudssMatrix_t x_mt;
+  cudssData_t cudss_data;
+  cudssHandle_t handle = at::cuda::getCurrentCudssHandle();
+
+  cudssConfigCreate(&config);
+  cudssDataCreate(handle, &cudss_data);
+
+  AT_DISPATCH_FLOATING_TYPES(values.type(), "create_matrix", ([&] {
+    scalar_t* values_ptr = values.data<scalar_t>();
+    scalar_t* b_ptr = b.data<scalar_t>();
+    scalar_t* x_ptr = x.data<scalar_t>();
+    auto CUDA_R_TYP = std::is_same<scalar_t, double>::value ? CUDA_R_64F : CUDA_R_32F;
+    cudssMatrixCreateDn(&b_mt, b.size(0), 1, b.size(0), b_ptr, CUDA_R_TYP, CUDSS_LAYOUT_COL_MAJOR);
+    cudssMatrixCreateDn(&x_mt, x.size(0), 1, x.size(0), x_ptr, CUDA_R_TYP, CUDSS_LAYOUT_COL_MAJOR);
+    cudssMatrixCreateCsr(&A_mt, A.size(0), A.size(1),  A._nnz(), rowOffsets, rowOffsets + crow.size(0), colIndices, values_ptr, CUDA_R_32I, CUDA_R_TYP, CUDSS_MTYPE_GENERAL, CUDSS_MVIEW_FULL, CUDSS_BASE_ZERO);
+  }));
+  cudssExecute(handle, CUDSS_PHASE_ANALYSIS, config, cudss_data, A_mt, x_mt, b_mt);
+  cudssExecute(handle, CUDSS_PHASE_FACTORIZATION, config, cudss_data, A_mt, x_mt, b_mt);
+  cudssExecute(handle, CUDSS_PHASE_SOLVE, config, cudss_data, A_mt, x_mt, b_mt);
```

Comment:
It looks like all of the struct creation and execute calls will have status returns that should be checked.

Reply:
fixed by https://github.com/pytorch/pytorch/pull/129856/commits/51614e9ccc67bfb214ea6f1f4605997fd8843d53